0000-catalog.md

```diff
@@ -0,0 +1,744 @@
+repeated field of **PropertyValues**.  The properties available are defined by 
+the Grid Property Schema Transaction Family and are restricted to the fields 
+and rules of the GS1 Catalog schema (which will be defined at a later time).  
+Transactions which are responsible for setting product state values must ensure 
+that those properties conform with the requirements of the GS1 Catalog Property 
+Schema (to be defined at a later time).
+
+``` 
+message Catalog { 
+    string catalog_id = 1; 
+    string owner = 2; 
+    string name = 3; 
+    repeated PropertyValue properties = 4; 
+} 
+```
+
+### Referencing a Catalog
+
+Catalogs are uniquely referenced by their catalog_id. For example:
+
+``` 
+get_catalog(catalog_id) // hash(catalog_name) 
+set_catalog(catalog_id, catalog) // hash(catalog_name), catalog 
+```
+
+### Catalog Addressing in the Merkle-Radix State System
+
+In order to uniquely locate Catalogs in the Merkle-Radix state system, an
+address must be constructed which identifies the storage location of the Catalog
+representation.
+
+All Grid addresses are prefixed by the 6-hex-character namespace prefix
+"621dee",  Catalogs are further prefixed under the Grid namespace with reserved
+enumerations of "03" ("00", "01", and "02" being reserved for other purposes)
+indicating "Grid Catalogs" and an additional org_id. Currently in Grid Product, 
+org_ids are a [6-10 digit GS1 company prefix](https://www.gs1-us.info/gs1-company-prefix/). 
+To keep the prefix length consistent, any GS1 company prefix less than 10 digits will 
+be left padded with the hexadecimal representation of the number 10 ("a") until it is 
+10 digits in length. This composite address will enable inexpensive catalog operations.
+
+Therefore, all addresses starting with:
+
+``` 
+"621dee" + "03" + "aaaa123456" // org_id (in this case a gs1_company_prefix)
+```
+
+are Grid GS1 Catalogs identified by the hash of the catalog name which can be
+referenced by a "catalog_product" to group products to a specific catalog. 
+
+The catalog_id format consists of 15-digit "alphanumeric string" which include 
+a fixed amount of internal "0" padding.  After the 18-hex-characters that are 
+consumed by the grid namespace prefix, the catalog, and gs1_company_prefix there 
+are 52 hex characters remaining in the address.  The 15 digits of the catalog_id can 
+be left padded with 60-hex-character zeroes and right padded with 
+2-hex-character zeroes to accommodate potential future storage associated with 
+the GS1 Catalog representation, for example:
+
+``` 
+"621dee" + "03" + "123456" + "000000000000000000000000000000000000000" +
+15-character "numeric string" catalog_id + "00" // catalog_id == hash(catalog_name) 
+```
+
+Using SHA-2 from the rust-crypto crate we can tie the catalog_id to 
+catalog_name to prevent duplication of catalogs when invoking the 
+catalog_create action.  
+```
+extern crate crypto;
+use self::crypto::digest::Digest;
+use self::crypto::sha3::Sha3;
```

Comment:
Current code (pike, product) uses crypto::sha2::Sha512. We should probably stick to the same one.

Reply:
✅